### PR TITLE
Buildings and roads

### PR DESCRIPTION
**Pull request by taihak94**

### CODE REVIEW

File: analona.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    def run_schema(self, data):
+        return(self.schema.validate(data))
+
+    def validate_schema(self):
+        try:
+            self.run_schema(self.item)
+        except SchemaError as x:
+            return x.code
+        else:
+            return self.item
```
Comment (mzedayda) [inline analona.py]: Why not 'true' ?

File: analona.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+
+class BaseItem(Validator):
```
Comment (mzedayda) [inline analona.py]: Object

File: analona.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    
+    def get_basic_schema(self):
+        return ({
+            '_id': Or(str, Use(int)),
+            'company': str,
+            'geometry': {
+                'coordinates': list,
+                'type': Or("Point", "Polygon", error="geometry type: should be Point or Polygon")
+            },
+            'sourceItem': And(str, len),
```
Comment (mzedayda) [inline analona.py]: ?

File: analona.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        return ({})
+
+    def validate(self):
+        return self.validate_item()
+
+def is_list_of_string(ids):
+    is_list = True
+    for item in ids:
+        is_list = isinstance(item, str)
+    return is_list
```
Comment (mzedayda) [inline analona.py]: Check all of them, not only the last

File: analona.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+
+class Ship(BaseItem):
+    """
+    Ship validator
+    """
+    def __init__(self, ship):
+        BaseItem.__init__(self, ship, self.create_extended_schema())
+
+    def create_extended_schema(self):
```
Comment (mzedayda) [inline analona.py]: Have only 'schema' and 'validate' methods for each object

File: analona.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+class Ship(BaseObject):
+    """
+    Ship validator
+    """
+    def __init__(self, ship):
+        BaseObject.__init__(self, ship)
+    
+    def compose_schema(self):
+        schema = self.get_schema()
```
Comment (mzedayda) [inline analona.py]: This belongs to 'get_schema', no?

I mean:
```
def get_schema(self):
        base_schema = BaseObject.get_schema()
        additional...
        schema = update...
        return schema
```

File: analona.py [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+        
+        return ({
+            '_id': Or(str, Use(int)),
+            'company': str,
+            Optional('geometry'): {
+                'coordinates': list,
+                'type': Or("Point", "Polygon", "MultiPolygon", "MultiPoint", error="geometry type error")
+            },
+            'provisionTime': Use(parser.parse),
+            Optional('analyticsUrl'): Or(
```
Comment (mzedayda) [inline analona.py]: URL is sufficient? platform? creds? other stuff?

File: analona.py (hunk 6 above)
Comment (taihak94) [reply analona.py]: if I do that i'll override the get_schema method from the parent...

File: analona.py [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        url_regex = r'https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)'
+        
+        return ({
+            '_id': Or(str, Use(int)),
+            'company': str,
+            Optional('geometry'): {
+                'coordinates': list,
+                'type': Or("Point", "Polygon", "MultiPolygon", "MultiPoint", error="geometry type error")
+            },
+            'provisionTime': Use(parser.parse),
```
[the remark this answers is not in the document]
Comment (taihak94) [reply analona.py]: when did we get it

File: analona.py (hunk 6 above)
Comment (mzedayda) [reply analona.py]: And that's a bad thing because?

**Pull request merged**